presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSkipEmptyFiles.java

```diff
@@ -0,0 +1,360 @@
+                .build();
+
+        this.queryBucketFailRunner.createCatalog(CATALOG, CATALOG, properties);
+        this.queryBucketFailRunner.execute(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
+    }
+
+    /**
+     * Generates a temporary directory and creates two parquet files inside, one is empty and the other is not
+     *
+     * @param tableName a {@link String} containing the desired table name
+     */
+    private void generateMetadata(String tableName)
+            throws Exception
+    {
+        Path tempDirectory = Files.createDirectory(Paths.get(temporaryDirectory.getPath(), tableName));
+        Path firstParquetFile = createTempFile(tempDirectory, randomUUID().toString(), randomUUID().toString());
+        ParquetTester.writeParquetFileFromPresto(firstParquetFile.toFile(),
+                ImmutableList.of(IntegerType.INTEGER),
+                Collections.singletonList("field"),
+                new Iterable[] {Collections.singleton(1)},
+                1,
+                GZIP,
+                PARQUET_2_0);
+        createTempFile(tempDirectory, randomUUID().toString(), randomUUID().toString());
+    }
+
+    /**
+     * Generates a temporary directory and inserts data in every partition of the bucketed table, including an empty file in the first partition
+     *
+     * @param queryRunner a {@link QueryRunner} with the desired configuration properties
+     * @param tableName a {@link String} containing the desired table name
+     * @param replaceDataFileByEmptyFile a {@code true} if it is necessary to delete a partition file
+     */
+    private void generateBucketedMetadataWithEmptyFiles(DistributedQueryRunner queryRunner, String tableName, boolean replaceDataFileByEmptyFile) throws Exception
+    {
+        Path tempDirectory = Files.createDirectory(Paths.get(temporaryDirectory.getPath(), tableName));
+        @Language("SQL") String createQuery = format("CREATE TABLE %s.\"%s\".\"%s\" (id %s, field %s) WITH (external_location = '%s'," +
+                        "format = 'Parquet',partitioned_by = ARRAY['field']," +
+                        "  bucketed_by = ARRAY['id']," +
+                        "  bucket_count = 3)",
+                CATALOG, SCHEMA, tableName, IntegerType.INTEGER, VarcharType.VARCHAR, tempDirectory.toUri());
+        queryRunner.execute(createQuery);
+        Path partitionDirectory = Paths.get(tempDirectory + "/field=field1");
+        String partitionName = "field";
+        @Language("SQL") String insertQuery;
+        for (int i = 1; i <= 5; i++) {
+            insertQuery = format("INSERT INTO %s.\"%s\".\"%s\" VALUES (%s,'%s')",
+                    CATALOG, SCHEMA, tableName, i, partitionName + i);
+            queryRunner.execute(insertQuery);
+        }
+        if (replaceDataFileByEmptyFile) {
+            Files.delete(Arrays.stream(requireNonNull(partitionDirectory.toFile().listFiles((dir, name) -> !name.endsWith(".crc")))).iterator().next().toPath());
```

Comment:
Can we break this up into multiple lines for readability?

Reply:
Done